src/Integration.Vsix.UnitTests/Analysis/SingleFileMonitorTests.cs

```diff
@@ -381,85 +378,80 @@ public void Real_File_ChangesToOtherFiles_AreIgnored()
+        [TestMethod]
+        public void File_DuplicateChangesDifferentTime_NotIgnored()
+        {
+            var filePath = "test\\";
 
-            TestContext.WriteLine($"Number of recorded events: {totalEventCount}");
-            totalEventCount.Should().BeInRange(340, 500);
+            var fileSystemMock = CreateFileSystemMock();
 
-            void testBody()
-            {
-                const int pauseBetweenOpsInMs = 125;
+            var watcherFactoryMock = CreateFactoryAndWatcherMocks(out var watcherMock);
+            var testLogger = new TestLogger(logToConsole: true, logThreadId: true);
 
-                using (var singleFileMonitor = new SingleFileMonitor(filePathToMonitor, testLogger))
-                {
-                    var testWrapper = new WaitableFileMonitor(singleFileMonitor);
-                    testWrapper.EventCount.Should().Be(0);
+            var dateTime = DateTime.Now;
 
-                    // 1. Create the file
-                    File.WriteAllText(filePathToMonitor, "initial text");
-                    testWrapper.PauseForEvent(pauseBetweenOpsInMs);
+            using (var fileMonitor = new SingleFileMonitor(watcherFactoryMock.Object, fileSystemMock.Object, filePath, testLogger))
+            {
+                var waitableFileMonitor = new WaitableFileMonitor(fileMonitor);
+                waitableFileMonitor.EventCount.Should().Be(0);
+
+                fileSystemMock.Setup(x => x.File.GetLastWriteTimeUtc(filePath)).Returns(dateTime);
+                watcherMock.Raise(x => x.Changed += null, new FileSystemEventArgs(WatcherChangeTypes.Changed, filePath, ""));
 
-                    // 2. Amend the file
-                    File.AppendAllText(filePathToMonitor, " more text");
-                    testWrapper.PauseForEvent(pauseBetweenOpsInMs);
+                fileSystemMock.Setup(x => x.File.GetLastWriteTimeUtc(filePath)).Returns(dateTime.AddMilliseconds(1));
+                watcherMock.Raise(x => x.Changed += null, new FileSystemEventArgs(WatcherChangeTypes.Changed, filePath, ""));
 
-                    // 3. Rename from the tracked name
-                    var renamedFile = Path.ChangeExtension(filePathToMonitor, "moved");
-                    File.Move(filePathToMonitor, renamedFile);
-                    testWrapper.PauseForEvent(pauseBetweenOpsInMs);
+                waitableFileMonitor.EventCount.Should().Be(2);
+            }
+        }
 
-                    // 4. Rename back to the tracked name
-                    File.Move(renamedFile, filePathToMonitor);
-                    testWrapper.PauseForEvent(pauseBetweenOpsInMs);
+        private Mock<IFileSystem> CreateFileSystemMock()
+        {
+            var fileSystemMock = new Mock<IFileSystem>();
+            fileSystemMock.Setup(x => x.Directory.Exists(It.IsAny<string>())).Returns(true);
+            fileSystemMock.Setup(x => x.File.GetLastWriteTimeUtc(It.IsAny<string>())).Returns(DateTime.MaxValue);
```

Comment:
is this needed?

Reply:
Yes, else some tests have an issue in SingleFileMonitor.cs  where that line is called